wifi-subsys/components/network/src/WiFi.c

```diff
@@ -0,0 +1,261 @@
+{
+   esp_err_t err;
+   err = nvs_set_string(NVS_APP_NAMESPACE, USER_AP_NAME_KEY, AP_WIFI_SSID);
+   if(err != ESP_OK){
+       ESP_LOGE(TAG, "Error: to setting AP SSID to the nvs");
+   }
+
+   err = nvs_set_string(NVS_APP_NAMESPACE, USER_AP_PASSWORD_KEY, AP_WIFI_PASSWORD);
+   if(err != ESP_OK){
+       ESP_LOGE(TAG, "Error: to setting AP Password to the nvs");
+   }
+
+   err = nvs_set_string(NVS_APP_NAMESPACE, USER_STA_NAME_KEY, STA_WIFI_SSID);
+   if(err != ESP_OK){
+       ESP_LOGE(TAG, "Error: to setting AP Password to the nvs");
+   }
+
+   err = nvs_set_string(NVS_APP_NAMESPACE, USER_STA_PASSWORD_KEY, STA_WIFI_PASSWORD);
+   if(err != ESP_OK){
+       ESP_LOGE(TAG, "Error: to setting AP Password to the nvs");
+   }
+    /* cppcheck-suppress duplicateCondition
+    * (reason: <your reason why you think this is a false positive>) */
+   if(err != ESP_OK){
+       nvs_set_uint8(NVS_APP_NAMESPACE, NVS_IS_CONFIGURED_KEY, 0);
+       return err;
+   } else {
+       nvs_set_uint8(NVS_APP_NAMESPACE, NVS_IS_CONFIGURED_KEY, 1);
+       return ESP_OK;
+   }
+
+   return ESP_OK;
+}
+
+esp_err_t wifi_init(void)
+{
+
+    ESP_LOGI(TAG, "starting wifi module");
+
+    if(nvs_init() != ESP_OK){
+        ESP_LOGE(TAG, "Error: starting nvs");
+    }
+
+    ESP_ERROR_CHECK(esp_netif_init());
+
+    ESP_ERROR_CHECK(esp_event_loop_create_default());
+
+    esp_netif_create_default_wifi_sta();
+    esp_netif_create_default_wifi_ap();
+
+    wifi_init_config_t cfg = WIFI_INIT_CONFIG_DEFAULT();
+
+    ESP_ERROR_CHECK(esp_wifi_init(&cfg));
+
+    esp_event_handler_instance_t instance_any_id;
+    esp_event_handler_instance_t instance_got_ip;
+
+    ESP_ERROR_CHECK(esp_event_handler_instance_register(WIFI_EVENT,
+                                                        ESP_EVENT_ANY_ID,
+                                                        &event_handler,
+                                                        NULL,
+                                                        &instance_any_id));
+
+    ESP_ERROR_CHECK(esp_event_handler_instance_register(IP_EVENT,
+                                                        IP_EVENT_STA_GOT_IP,
+                                                        &event_handler,
+                                                        NULL,
+                                                        &instance_got_ip));
+
+
+    int is_configured = wifi_credentials_are_configured();
+
+
+    if(!is_configured){
+        set_default_credentials();
+    }
+
+    #if WIFI_ENABLE_STA && WIFI_ENABLE_AP
+    ESP_ERROR_CHECK(esp_wifi_set_mode(WIFI_MODE_APSTA));
+    ESP_ERROR_CHECK(init_sta());
+    ESP_ERROR_CHECK(init_ap());
+    return ESP_OK;
+    #endif
+
+    #if CONFIG_ENABLE_AP
+    ESP_ERROR_CHECK(esp_wifi_set_mode(WIFI_MODE_AP));
+    ESP_ERROR_CHECK(init_ap());
+    #endif
+
+    #if CONFIG_ENABLE_STA
+    ESP_ERROR_CHECK(esp_wifi_set_mode(WIFI_MODE_STA));
+    ESP_ERROR_CHECK(init_sta());
+    #endif
+
+    return ESP_OK;
+}
```

Comment:
new line at EOF

Reply:
<img width="511" alt="Screen Shot 2022-01-27 at 4 49 38 PM" src="https://user-images.githubusercontent.com/37220428/151441381-66b646cc-91b2-4d22-b382-a1bb80a7077e.png">

this was already added

Reply:
![Captura de pantalla de 2022-01-27 21-58-14](https://user-images.githubusercontent.com/10358374/151442580-67a62661-aa3a-45d7-8d9b-fa290f02c449.png)

but not syncronized between your pc and the repo for some reason